cf/mixin/fielddomain.py

```diff
@@ -1207,13 +1207,15 @@ def autocyclic(self, key=None, coord=None, verbose=None, config={}):
             bounds_units = bounds.Units
 
         period = coord.period()
+
         if period is not None:
             has_period = True
         else:
             period = config.get("period")
-            has_period = False
+            if period is None:
+                has_period = False
```

Comment:
By the logic here I think there can be a case where we could run into a `NameError` because `has_period` would not be defined (if `config.get("period")` is not `None`?), so to protect against that, I would add something like:

```suggestion
            if period is None:
                has_period = False
            else:
                has_period = True
              
```

Reply:
Good spot